### PR TITLE
Add help description for remote build

### PR DESCRIPTION
**Pull request by Hazhzeng**
We add a description for --build flag

### CODE REVIEW

**Comment by ColbyTresness** [comment]
What about adding this to the list when the user runs func --help? This should show up as an option

File: src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs [hunk 1]
```diff
@@ -84,6 +84,7 @@ public override ICommandLineParserResult ParseArgs(string[] args)
             Parser
                 .Setup<BuildOption>('b', "build")
                 .SetDefault(BuildOption.Default)
+                .WithDescription("Perform build action when deploying Linux function apps. (accepts: remote)")
```
Comment (ahmedelnably) [inline src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs]: `when deploying to a Linux Function app....`

**Comment by Hazhzeng** [comment]
> What about adding this to the list when the user runs func --help? This should show up as an option

The CLI library (fluent-command-line-parser)'s --help context complies with **<command> <contexts> <actions>** hierarchy. Since the **--build** flag is under **azure functionapp publish** context, the help will only show up when user runs **func azure functionapp --help**.

**Comment by Hazhzeng** [comment]
Close-Open pull request to retrigger build

**Pull request merged**